src/widgets/styles/qstyleoption.h

```diff
@@ -207,8 +207,14 @@ class Q_WIDGETS_EXPORT QStyleOptionHeaderV2 : public QStyleOptionHeader
     bool isSectionDragTarget:1;
     int unused:29;
 
+    // Used where an unknown V1 or V2 instance needs to be copied for modification without risking object slicing
+    static QStyleOptionHeaderV2 copyFromV1OrV2(const QStyleOptionHeader *other);
```

Comment:
I could have just made the extra constructor below public here. But I wanted to define a named method for this action to emphasise that it's kind of an unusual way to use this.

The Qt fix for this in 6.5.1 was slightly different, and annoyingly copy-pasted in two places rather than defined on this class (possibly for binary compatibility reasons?):
https://codereview.qt-project.org/c/qt/qtbase/+/470791/2/src/widgets/styles/qcommonstyle.cpp